### PR TITLE
Remove unsafe FFI and monkey patches to OpenSSL

### PR DESCRIPTION
**Pull request by rickmark**
Monkey patching OpenSSL's Point function can lead to cryptographic errors when using curves other then `secp256k1` and also by loading OpenSSL both via FFI and `require 'openssl'` can cause the same binary to bind to multiple different OpenSSL libraries.

### CODE REVIEW

**Comment by justinsoong** [comment]
@rickmark 👍 for this PR, i think this is going to solve some down stream deps that require this PR to be merged, looks like no longer maintained?

**Comment by q9f** [comment]
Yeah, it would be great to get this released. Please let me know if I can be of any assistance!

**Comment by rickmark** [comment]
> @rickmark 👍 for this PR, i think this is going to solve some down stream deps that require this PR to be merged, looks like no longer maintained?

@coinbase maintains a fork, but I'm not sure how well its sync'd - you may have to ask them to upstream some changes

**Comment by wink** [comment]
Hey folks, I appreciate the pull request. It's nice to see this library still have life in projects, although it hasn't been actively maintained in a while. I'm open to bringing on someone to co-maintain it. I'll be chatting with @q9f about it after Thanksgiving.

**Comment by rickmark** [comment]
> Hey folks, I appreciate the pull request. It's nice to see this library still have life in projects, although it hasn't been actively maintained in a while. I'm open to bringing on someone to co-maintain it. I'll be chatting with @q9f about it after Thanksgiving.

I'm happy to do some review / merges if you need to divest from it as well

**Comment by wink** [comment]
Quick announcement: @q9f is now a co-maintainer on this project. Welcome, Afri! Thank you for picking up where I left off many years ago!

**Comment by q9f** [comment]
Thank you @wink!

@rickmark could you rebase this on latest master, so that we can get a functional CI? 🙏🏼

File: lib/money-tree/version.rb [hunk 1]
```diff
@@ -1,3 +1,3 @@
 module MoneyTree
-  VERSION = "0.10.0"
+  VERSION = "1.0"
```
Comment (q9f) [inline lib/money-tree/version.rb:2]: let me handle the version in #48

**Review by q9f — APPROVED** [review]

**Comment by q9f** [comment]
Thank you. This is good to go. I will create more tests in a subsequent PR to make sure, this is bulletproof. 👍🏼

**Pull request merged**